docs/src/evaluating.md

```diff
@@ -68,9 +68,95 @@ int status = await page.EvaluateAsync<int>(@"async () => {
 }");
 ```
 
+## Different environments
+
+Evaluated scripts run in the browser environment, while your test runs in a testing environments. This means you cannot use variables from your test in the page and vice versa. Instead, you should pass them explicitly as an argument.
+
+The following snippet is **WRONG** because it uses the variable directly:
+
+```js
+const data = 'some data';
+const result = await page.evaluate(() => {
+  // WRONG: there is no "data" in the web page.
+  window.myApp.use(data);
+});
+```
+
+```java
+String data = "some data";
+Object result = page.evaluate("() => {\n" +
```

Comment:
not sure if these snippets make much sense on languages beside JS/JS. Seems confusing if we speak about a variable which is available across a program language barrier.

Reply:
I kind of agree, but wasn't able to meaningfully improve this part.